python/foglamp/plugins/south/cc2650/cc2650.py

```diff
@@ -0,0 +1,403 @@
+    Returns:
+        new_handle: new handle to be used in the future calls
+    Raises:
+    """
+    # In this method, cannot use "handle" as it might have changed due to restart. Hence use "_handle".
+    global _handle
+
+    bluetooth_adr = _handle['bluetoothAddress']['value']
+    _LOGGER.info("Old config for CC2650 {} plugin {} \n new config {}".format(bluetooth_adr, _handle, new_config))
+
+    # Find diff between old config and new config
+    diff = utils.get_diff(_handle, new_config)
+
+    # Plugin should re-initialize and restart if key configuration is changed
+    if 'bluetoothAddress' in diff or \
+                'assetPrefix' in diff or \
+                'maxRetry' in diff or \
+                'temperatureSensor' in diff or \
+                'luminanceSensor' in diff or \
+                'humiditySensor' in diff or \
+                'pressureSensor' in diff or \
+                'movementSensor' in diff or \
+                'batteryData' in diff:
+        plugin_shutdown(_handle)
+        new_handle = plugin_init(new_config)
+        new_handle['restart'] = 'yes'
+        _LOGGER.info("Restarting CC2650 {} plugin due to change in configuration keys [{}]".format(bluetooth_adr, ', '.join(diff)))
+    elif 'pollInterval' in diff or 'connectionTimeout' in diff or 'shutdownThreshold' in diff:
+        new_handle = copy.deepcopy(new_config)
+        new_handle['restart'] = 'no'
+    else:
+        new_handle = copy.deepcopy(new_config)
+        new_handle['restart'] = 'no'
+    return new_handle
+
+
+def _plugin_stop(handle):
+    """ Stops the plugin doing required cleanup
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+    Raises:
+    """
+    if 'tag' in handle:
+        bluetooth_adr = handle['bluetoothAddress']['value']
+        tag = handle['tag']
+
+        if tag.is_connected:
+            # Disable sensors
+            tag.char_write_cmd(handle['characteristics']['temperature']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['luminance']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['humidity']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['pressure']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['movement']['configuration']['handle'], movement_disable)
+            tag.disconnect()
+        handle.pop('tag', None)
+        _LOGGER.info('SensorTagCC2650 {} stopped.'.format(bluetooth_adr))
+
+
+def plugin_shutdown(handle):
+    """ Shutdowns the plugin doing required cleanup, to be called prior to the South service being shut down.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+    Raises:
+    """
+    bluetooth_adr = handle['bluetoothAddress']['value']
+    _plugin_stop(handle)
```

Comment:
see my comment in internal def _plugin_stop which is not a standard hook def  of plugin

Reply:
yes, _plugin_stop() is not part of the plugin contract. It is an internal def. It is never used in the calling code of south/server.py.